### DISCUSSION TITLE
integration with testing-library : fireEvent

### DISCUSSION THREAD

**Discussion by zoutigo** [react-page/react-page#1098] [Q&A]
Hello.
I'm trying to write a test suite for my component.
However , i'don't think i'm writting it on the good way for react-page.

```
 <Editor
                data-testid="page-editor"
                {...field}
                cellPlugins={cellPlugins}
                cellSpacing={cellSpacing}
                lang="fr"
                onChange={(newvalue) => field.onChange(newvalue)}
              />
```

```
 fireEvent.change(screen.getByTestId('page-editor'), {
      target: { value: 'ceci est un test relatif' },
    })

    await waitFor(() => {
      expect(screen.getByText(/ceci est un test relatif/i)).toBeInTheDocument()
    })
  })
```
getByTestId is not working.
Then the fireEvent is not setting the value properly.

Do you please have an example of how to perform test with testing-library or something similar ?
Is it possible to put it in the doc ( a test example folder)

Thanks in advance.

**Comment by macrozone** [comment]
@zoutigo I don't know the exact problem but multiple observation:

- passing additional props to `<Editor />` like `data-testid` might not work, its a complex component and not just an input field
- its better to wrap the `Editor` in a div with a testid, that way you can subselect dom-elements from that div.
- firing a `change` event in the editor might not work, its not a text-input
- instead, you should mimick user interactions (clicks and typings)

Unfortunatly there is not yet much support from ReactPage in this regard, a reason for this is the high complexity.

I would appreciate any help in this regard though
